### PR TITLE
Feature: Optimize the replica delay api logic

### PR DESCRIPTION
**Pull request by mateng0915**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Optimize behavior for a replica delay api logic in case the replica is read-only.

Description:
============
Currently if in the Whole CK instance has one table is read only status then the API  /replicas_status will throw error, xxx table is read only and can not get other normal ReplicatedMergeTree table's delay value

Solution:
=========
Return other normal table's delay value even if the CK instance has readonly Replicatedxxx table

Please enter the commit message for your changes. Lines starting

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Server/ReplicasStatusHandler.cpp [hunk 1]
```diff
@@ -59,14 +59,22 @@ void ReplicasStatusHandler::handleRequest(HTTPServerRequest & request, HTTPServe
                 time_t absolute_delay = 0;
                 time_t relative_delay = 0;
 
-                table_replicated->getReplicaDelays(absolute_delay, relative_delay);
+                if(!table_replicated->isTableReadOnly())
```
Comment (pufit) [inline src/Server/ReplicasStatusHandler.cpp:62]: space before brackets

File: src/Server/ReplicasStatusHandler.cpp [hunk 2]
```diff
@@ -59,14 +59,22 @@ void ReplicasStatusHandler::handleRequest(HTTPServerRequest & request, HTTPServe
...
-                    || (settings.min_relative_delay_to_close && relative_delay >= static_cast<time_t>(settings.min_relative_delay_to_close)))
-                    ok = false;
+                    if ((settings.min_absolute_delay_to_close && absolute_delay >= static_cast<time_t>(settings.min_absolute_delay_to_close))
+                        || (settings.min_relative_delay_to_close && relative_delay >= static_cast<time_t>(settings.min_relative_delay_to_close)))
+                        ok = false;
+
+                    message << backQuoteIfNeed(db.first) << "." << backQuoteIfNeed(iterator->name())
+                        << ":\tAbsolute delay: " << absolute_delay << ". Relative delay: " << relative_delay << ".\n";
+
+                } else {
```
Comment (pufit) [inline src/Server/ReplicasStatusHandler.cpp:73]: `else` should be on a new line

**Review by pufit — CHANGES_REQUESTED** [review]

File: src/Server/ReplicasStatusHandler.cpp (hunk 1 above)
Comment (mateng0915) [reply src/Server/ReplicasStatusHandler.cpp]: Thx comment,  Fixed it

File: src/Server/ReplicasStatusHandler.cpp (hunk 2 above)
Comment (mateng0915) [reply src/Server/ReplicasStatusHandler.cpp]: Thx comment, Fixed it

**Review by pufit — APPROVED** [review]

**Comment by mateng0915** [comment]
@pufit  Hello pufit can you help merge this PR, thx

**Comment by pufit** [comment]
> @pufit Hello pufit, can you help merge this PR, thx

Yes, sure. Just wanted to run CI before merging. It's not that I've expected any bug in your commit, just a good practice in general.

**Pull request merged**